src/jenkins_job_insight/main.py

```
@@ -2544,18 +2542,88 @@ async def get_capabilities(settings: Settings = Depends(get_settings)) -> dict:
+class JiraProjectsRequest(BaseModel):
+    """Request body for listing Jira projects with user credentials."""
+
+    jira_token: str = Field(default="", description="User's Jira token")
+    jira_email: str = Field(default="", description="User's Jira email for Cloud auth")
+    query: str = Field(default="", description="Search query to filter projects")
+
+
+@app.post("/api/jira-projects")
 async def list_jira_projects(
+    body: JiraProjectsRequest,
     settings: Settings = Depends(get_settings),
```

Ruff (0.15.9)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)